crypto/bio/bio_mem.c

```diff
@@ -76,7 +76,7 @@ BIO *BIO_new_mem_buf(const void *buf, ossl_ssize_t len) {
   BUF_MEM *b;
   BIO_BUF_MEM *bbm;
 
-  const size_t size = (len < 0 || (size_t)len > SIZE_MAX) ? strlen((char *)buf) : (size_t)len;
+  const size_t size = len < 0 ? strlen((char *)buf) : (size_t)len;
 
   if (!buf && len != 0) {
```

Comment:
`strlen` is undefined for a NULL argument, no?

So, this check should be before the calculation of `size` above.